src/Voron/Data/CompactTrees/CompactTree.cs

```diff
@@ -532,15 +538,15 @@ private void Verify()
                     dictionary.Decode(encodedKey, ref decodedKey);
 
                     dictionary.Decode(lastEncodedKey, ref lastDecodedKey);
-                    Debug.Fail("");
+                    VoronUnrecoverableErrorException.Raise(_llt, "Last encoded key does not follow lexicographically.");
                 }
                 
                 lastEncodedKey = encodedKey;
                 lastDecodedKey = decodedKey;
             }
         }
 
-        private void MaybeMergeEntries(ref CursorState destinationState)
+        private bool MaybeMergeEntries(ref CursorState destinationState)
```

Comment:
Who is using the return value?

Reply:
This was useful for debugging. Should I kill it?

Reply:
I think so